synonyms.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import "errors"
+
+type Graph map[string]map[string]bool
```

Comment:
Is that bool type ever read/set?
Also: is the go-map thread-safe or do you wrap it somewhere?